rnd/autogpt_server/autogpt_server/data/agent_block.py

```diff
@@ -0,0 +1,144 @@
+    def get_commands(self) -> Iterator[Command]:
+        yield self.output
+    
+    @command(
+        parameters={
+            "output": JSONSchema(
+                type=JSONSchema.Type.STRING,
+                description="Output data to be returned.",
+                required=True,
+            ),
+        },
+    )
+    def output(self, output: str) -> str:
+        """Use this to output the result."""
+        return output
+
+
+class BlockAgent(Agent):
+    def __init__(
+        self,
+        settings: BlockAgentSettings,
+        llm_provider: MultiProvider,
+        file_storage: FileStorage,
+        app_config: AppConfig,
+    ):
+        super().__init__(settings, llm_provider, file_storage, app_config)
+
+        self.output = OutputComponent()
+
+        # Disable components
+        for attr_name in list(self.__dict__.keys()):
+            attr_value = getattr(self, attr_name)
+            if type(attr_value).__name__ in settings.disabled_components:
+                delattr(self, attr_name)
+
+
+class AutoGPTOutput(BaseModel):
+    result: str
+    tool: str
+
+class AutoGPTAgentBlock(Block):
+    class Input(BlockSchema):
+        task: str
+        input: str
+        disabled_components: list[str] = Field(default_factory=list)
+        disabled_commands: list[str] = Field(default_factory=list)
+    
+    class Output(BlockSchema):
+        output: AutoGPTOutput
+        # error: str
+
+    def __init__(self):
+        super().__init__(
+            id="d2e2ecd2-9ae6-422d-8dfe-ceca500ce6a6",
+            input_schema=AutoGPTAgentBlock.Input,
+            output_schema=AutoGPTAgentBlock.Output,
+        )
+
+    def run(self, input_data: Input) -> BlockOutput:
+        # Set up configuration
+        config = ConfigBuilder.build_config_from_env()
+        # Disable commands
+        config.disabled_commands.extend(input_data.disabled_commands)
+
+        # Storage
+        local = config.file_storage_backend == FileStorageBackendName.LOCAL
+        restrict_to_root = not local or config.restrict_to_workspace
+        file_storage = get_storage(
+            config.file_storage_backend,
+            root_path=Path("data"),
+            restrict_to_root=restrict_to_root,
+        )
+        file_storage.initialize()
+
+        # LLM provider
+        multi_provider = MultiProvider()
+        for model in [config.smart_llm, config.fast_llm]:
+            # Ensure model providers for configured LLMs are available
+            multi_provider.get_model_provider(model)
+
+        # State
+        state = BlockAgentSettings(
+            agent_id="TemporaryAgentID",
+            name="WrappedAgent",
+            description="Wrapped agent for the Agent Server.",
+            task=f"Your task: {input_data.task}\n"
+                 f"Input data: {input_data.input}",
+            disabled_components=input_data.disabled_components,
+        )
+        # Disable slow models
+        state.config.big_brain = False
```

Comment:
GPT 3.5 tends to use non-existent tools and GPT 4 often skips model fields but at least this way it's faster

Reply:
This is now user configurable on the block and smart by default